services/API-service/src/api/event/event.service.ts

```diff
@@ -270,24 +270,26 @@ export class EventService {
     countryCodeISO3: string,
     disasterType: DisasterType,
     eventName: string,
-    date: Date,
+    lastUploadDate: LastUploadDateDto,
   ): Promise<void> {
     const countryAdminAreaIds =
       await this.getCountryAdminAreaIds(countryCodeISO3);
+
+    const uploadCutoffMoment = this.helperService.getUploadCutoffMoment(
```

Comment:
nice. I did this same refactor in my ongoing branch. So we align (but might have to solve some merge conflicts later).